### PR TITLE
[ci] Install bazel & run C++ sanity check for MacOS wheel validation on release automation

### PR DESCRIPTION
**Pull request by khluu**
Followup PR to add Bazel & C++ sanity check to #43130 
## Changes
- Add installation for bazel `5.4.1` in script
- Run C++ sanity check on MacOS wheels
- Cleanup bazel upon exit

## Checks
https://buildkite.com/ray-project/release-automation/builds/49 passes for all wheels on both Python & C++

### CODE REVIEW

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 1]
```diff
@@ -44,17 +65,23 @@ run_sanity_check() {
...
     )
+    (
+        cd release/util
+        bash sanity_check_cpp.sh
+    )
     conda deactivate
 }
 
 _clean_up() {
+    bazel clean
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:76]: check if bazel binary exists (as the `install_bazel` might fail), if `bazel` does not exist, do not run `bazel clean`, or maybe just `bazel clean || true`

Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed! Added `|| true` so the script doesn't exit if bazel is not found there.

**Comment by khluu** [comment]
I rebased against `master` and this is ready for another review.

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 2]
```diff
@@ -44,10 +65,15 @@ run_sanity_check() {
         cd release/util
         python sanity_check.py
     )
+    (
+        cd release/util
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:69]: remove these 3 lines? we do not need to `cd release/util` twice.

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 3]
```diff
@@ -16,6 +17,26 @@ fi
...
+    if [ "${mac_architecture}" = "arm64" ]; then
+      url="https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-darwin-arm64"
+    elif [ "${mac_architecture}" = "x86_64" ]; then
+      url="https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-darwin-amd64"
+    else
+      echo "Could not find matching bazelisk URL for Mac ${mac_architecture}"
+      exit 1
+    fi
+
+    target="$TMP_DIR/bin/bazel"
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:36]: again: caps for bash variables.

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 4]
```diff
@@ -16,6 +17,26 @@ fi
 mac_architecture=$1 # First argument is the architecture of the machine, e.g. x86_64, arm64
 export RAY_VERSION="${RAY_VERSION:-2.9.1}"
 export RAY_HASH="${RAY_HASH:-cfbf98c315cfb2710c56039a3c96477d196de049}"
+export USE_BAZEL_VERSION="${USE_BAZEL_VERSION:-5.4.1}"
+
+install_bazel() {
+    mkdir -p "$TMP_DIR/bin"
+    # Add bazel to the path.
+    export PATH="$TMP_DIR/bin:$PATH"
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:25]: this PATH thing should run before running the scripts? or right after making the temp dir. this is not really part of `install_bazel`.

**Review by aslonnie — COMMENTED** [review]
mostly good, just some small things.

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 2 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed!

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 3 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 4 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed

**Comment by khluu** [comment]
Triggered another test run: https://buildkite.com/ray-project/release-automation/builds/52

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 5]
```diff
@@ -16,6 +17,22 @@ fi
 mac_architecture=$1 # First argument is the architecture of the machine, e.g. x86_64, arm64
 export RAY_VERSION="${RAY_VERSION:-2.9.1}"
 export RAY_HASH="${RAY_HASH:-cfbf98c315cfb2710c56039a3c96477d196de049}"
+export USE_BAZEL_VERSION="${USE_BAZEL_VERSION:-5.4.1}"
+
+install_bazel() {
+    if [ "${mac_architecture}" = "arm64" ]; then
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:23]: nit: use double bracket `[[ ]]` here and on line 25

File: .buildkite/release-automation/verify-macos-wheels.sh [hunk 6]
```diff
@@ -43,19 +60,26 @@ run_sanity_check() {
...
     conda deactivate
 }
 
 _clean_up() {
+    bazel clean || true
     rm -rf "$TMP_DIR"
 }
 
 # Create tmp directory unique for the run
 TMP_DIR=$(mktemp -d "$HOME/tmp.XXXXXXXXXX")
```
Comment (aslonnie) [inline .buildkite/release-automation/verify-macos-wheels.sh:74]: also, please double quote `$(mktemp ...)`, as into `"$(mktemp ...)"`

**Review by aslonnie — APPROVED** [review]
just some nits:

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 5 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed

File: .buildkite/release-automation/verify-macos-wheels.sh (hunk 6 above)
Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: fixed

Comment (khluu) [reply .buildkite/release-automation/verify-macos-wheels.sh]: I don't think the new format fixes change any behavior but here is another run on BK: https://buildkite.com/ray-project/release-automation/builds/60

**Pull request merged**